gobblin-core/src/main/java/org/apache/gobblin/writer/partitioner/TimeBasedORCSerDeWriterPartitioner.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.writer.partitioner;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.util.ForkOperatorUtils;
+import org.apache.hadoop.hive.ql.io.orc.OrcSerde;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StandardStructObjectInspector;
+
+
+/**
+ * A {@link TimeBasedWriterPartitioner} for {@link OrcSerde.OrcSerdeRow}s.
+ *
+ * The {@link OrcSerde.OrcSerdeRow} that contains the timestamp can be specified using
+ * {@link TimeBasedORCSerDeWriterPartitioner#WRITER_PARTITION_COLUMN_NAME}.
+ * only accept simple value, e.g., "timestamp" or "event_date".
+ * The {@link TimeBasedORCSerDeWriterPartitioner#WRITER_PARTITION_COLUMN_INDEX} that contains the index of column, has better performance.
+ * The {@link TimeBasedORCSerDeWriterPartitioner#WRITER_PARTITION_COLUMN_PATTERN} that contains format of column when you have a date_time as string.
+ * format use {@link DateTimeFormatter} on default UTC, e.g., "yyyy-MM-dd HH:mm:ss:SSS"
+ * The {@link TimeBasedORCSerDeWriterPartitioner#WRITER_PARTITION_COLUMN_TIMEZONE} that contains timezone of column to convert date_time.
+ *
+ *
+ * If a record contains none of the specified fields, or if no field is specified, the current timestamp will be used.
+ */
+public class TimeBasedORCSerDeWriterPartitioner extends TimeBasedWriterPartitioner<Object> {
+
+  private static final String WRITER_PARTITION_COLUMN_NAME = ConfigurationKeys.WRITER_PREFIX + ".partition.column";
+  private static final String WRITER_PARTITION_COLUMN_INDEX =
+      ConfigurationKeys.WRITER_PREFIX + ".partition.column.index";
+  private static final String WRITER_PARTITION_COLUMN_PATTERN =
```

Comment:
Can these configurationKeys (pattern, format, etc.) shared by both Avro and ORC?